lib/braid/mirror.rb

```diff
@@ -176,56 +210,79 @@ def fetch_base_revision_if_missing
       end
     end
 
+    sig {void}
     def fetch
       git_cache.fetch(url) if cached?
       git.fetch(remote)
     end
 
+    sig {returns(T::Boolean)}
     def cached?
       git.remote_url(remote) == cached_url
     end
 
+    sig {returns(String)}
     def base_revision
-      if revision
+      # Avoid a Sorbet "unreachable code" error.
+      # TODO (typing): Is the revision expected to be non-nil nowadays?  Can we
```

Comment:
Yes revision and url are required to be non-null. However, eyeballing the code seems to suggest we don't enforce this ... which I am curious about. However it is marked as required in schema. Hmmm.

We should be able to delete the inferred_revision branch (and the inferred_revision method) but it is difficult to verify without digging into the code more than I have time to do right now